src/throttler.guard.ts

```diff
@@ -45,6 +59,8 @@ export class ThrottlerGuard implements CanActivate {
       this.commonOptions = {
         skipIf: this.options.skipIf,
         ignoreUserAgents: this.options.ignoreUserAgents,
+        getTracker: this.options.getTracker || this.getTracker,
+        generateKey: this.options.generateKey || this.generateKey,
       };
```

Comment:
```suggestion
      this.commonOptions = {
        skipIf: this.options.skipIf,
        ignoreUserAgents: this.options.ignoreUserAgents,
      };
      this.commonOptions.getTracker ??= this.getTracker;
      this.commonOptions.generateKey ??= this.generateKey
```

Reply:
I slightly modified your suggestion, so that the defaults are set afterwards if the `commonOptions` are not defined but the override is still possible.
The idea was to also allow overriding both methods via the module options even if no overrides at the throttler definition level exist.
Would that be a viable option for you or do you think it's better without this possibility?

Reply:
Looks like you got exactly what I was going for. I removed too much in my immediate suggestion. Nice catch :)